library/alloc/src/vec/spec_from_elem.rs

```diff
@@ -59,3 +59,18 @@ impl SpecFromElem for u8 {
         }
     }
 }
+
+impl SpecFromElem for bool {
+    #[inline]
+    fn from_elem<A: Allocator>(elem: bool, n: usize, alloc: A) -> Vec<bool, A> {
+        if elem == false {
+            return Vec { buf: RawVec::with_capacity_zeroed_in(n, alloc), len: n };
+        }
+        unsafe {
+            let mut v = Vec::with_capacity_in(n, alloc);
+            ptr::write_bytes(v.as_mut_ptr(), 1u8, n);
```

Comment:
```suggestion
            ptr::write_bytes(v.as_mut_ptr(), elem as u8, n);
```

I find this more fitting given the context of the other functions in this file. This way it also removes the unexplained 1u8.